### PR TITLE
Transliterate button

### PR DESCRIPTION
**Pull request by TheoChevalier**
Adding changes on top of #754 to add the button on entity view, so that we ship it in today’s release :)  Will comment when it’s ready for review

### CODE REVIEW

**Comment by TheoChevalier** [comment]
Removed secureText, because we’re already using it here https://github.com/mozfr/transvision/pull/792/files#diff-9072e93e713a4782f5407ac9e3dbc369R34

However, we don’t seem to be using it in ShowResults, I assume we should?

**Comment by TheoChevalier** [comment]
Ready for review, I’ll manually squash then merge without auto-squashing, so that we keep two commits

File: app/classes/Transvision/ShowResults.php [hunk 1]
```diff
@@ -296,22 +312,47 @@ public static function resultsTable($search_object, $search_results, $page)
                 $target_string2 = '';
             }
 
+            $string_id = md5($key . mt_rand());
+            $regular_string_id = 'string_' . $string_id;
+
+            /*
+                Find if we need to transliterate the string.
+                The string gets transliterated if the target local is serbian,
```
Comment (flodolo) [inline app/classes/Transvision/ShowResults.php]: nit: Serbian

File: app/classes/Transvision/ShowResults.php [hunk 2]
```diff
@@ -145,6 +145,22 @@ public static function getTranslationMemoryResults($strings, $search, $max_resul
...
+     * @param string $locale The target locale
+     *
+     * @return string The string transliterated into the target locale
+     */
+    public static function getTransliteratedString($string, $locale)
+    {
+        $request = new API(parse_url(API_ROOT . "transliterate/$locale/$string"));
+        $json = include MODELS . 'api/transliterate.php';
+
+        return $json[0];
```
Comment (flodolo) [inline app/classes/Transvision/ShowResults.php]: I still think this should check if $json actually has an element, and figure out what to do if the string returned is empty.

File: app/inc/constants.php [hunk 3]
```diff
@@ -17,6 +17,7 @@
 define('CACHE_ENABLED', isset($_GET['nocache']) ? false : true);
 define('CACHE_PATH',    INSTALL_ROOT . 'cache/');
 define('APP_SCHEME',    isset($_SERVER['HTTPS']) && $_SERVER['HTTPS'] != 'off' ? 'https://' : 'http://');
+define('API_ROOT',  APP_SCHEME . htmlspecialchars($_SERVER['HTTP_HOST'], ENT_QUOTES, 'UTF-8') . '/api/v1/');
```
Comment (flodolo) [inline app/inc/constants.php]: Follow existing alignment?

**Comment by flodolo** [comment]
Can't find anything wrong during testing. I feel like we should be more cautious with errors, especially considering the transliterate function requires a component that is not part of default PHP installations.

**Comment by TheoChevalier** [comment]
If last commit looks good, I’m ready to squash & merge

File: app/classes/Transvision/ShowResults.php [hunk 4]
```diff
@@ -145,6 +145,22 @@ public static function getTranslationMemoryResults($strings, $search, $max_resul
...
+     * @param string $locale The target locale
+     *
+     * @return string The string transliterated into the target locale
+     */
+    public static function getTransliteratedString($string, $locale)
+    {
+        $request = new API(parse_url(API_ROOT . "transliterate/$locale/$string"));
+        $json = include MODELS . 'api/transliterate.php';
+
+        return isset($json[0]) ? $json[0] : '@@missing@@';
```
Comment (flodolo) [inline app/classes/Transvision/ShowResults.php]: Call to this method is after checks on `@@missing@@`.  Let's either use an empty string, or something like 'Function not available'.

**Pull request merged**